src/internal/_isThenable.js

```diff
@@ -1,6 +1,9 @@
+var _hasMethod = require('./_hasMethod');
+
+
 /**
  * Tests if a value is a thenable (promise).
  */
 module.exports = function _isThenable(value) {
-  return (value != null) && (value === Object(value)) && typeof value.then === 'function';
```

Comment:
> I think that check was copied from underscore?

Possibly, but it could have come from many places.  You asked about it in #573, and it's been in there since #524.  Hey, @yuri, any recollection of this?